### PR TITLE
Optimization

### PR DESCRIPTION
**Pull request by beru**
Hi, this PR mainly optimizes AVX backend.

### CODE REVIEW

File: tiny_dnn/io/mnist_parser.h [hunk 1]
```diff
@@ -108,10 +108,11 @@ inline void parse_mnist_labels(const std::string& label_file, std::vector<label_
     if (magic_number != 0x00000801 || num_items <= 0)
         throw nn_error("MNIST label-file format error");
 
+    labels->resize(num_items);
     for (uint32_t i = 0; i < num_items; i++) {
         uint8_t label;
         ifs.read((char*) &label, 1);
-        labels->push_back((label_t) label);
+        (*labels)[i] = (label_t) label;
```
Comment (edgarriba) [inline tiny_dnn/io/mnist_parser.h]: safe type cast

**Comment by edgarriba** [comment]
@beru any idea about Appveyor error?

**Comment by beru** [comment]
> any idea about Appveyor error?

@edgarriba I have no idea. Probablly it timed out? Can you re-run it again?

**Comment by edgarriba** [comment]
@beru not sure I have full permission to appveyor. I've updated my settings but I think you should force a new push to see the results.

**Comment by beru** [comment]
@edgarriba It looks like error number in this message `Command exited with code -1073740940` means heap corruption.
http://stackoverflow.com/questions/11881911/error-msb3073-the-command-lib-tests-exe-vcend-exited-with-code/11883845#11883845

I'll check if it reproduces again with new commit.

**Pull request merged**